metabase/metabase
# Refactor model persistent settings FE

This PR addresses feedback on #21546. The original PR was using admin setting widgets to call API endpoints. And I've missed that there is an `onChanged` callback exactly for cases like that. 

Couldn't completely get rid of the custom widget for the "refresh interval" setting though. The `SettingsRadio` widget doesn't play nice with numeric option values and it doesn't actually look exactly as we want it to be. I believe there is still a way to extend and use `SettingsRadio`, but it seems risky atm

### To Verify

1. Open `/admin/settings/caching`
2. Ensure you can enable/disable caching for models
    * ensure `POST /api/persist/enable` is sent when you enable caching
    * ensure `POST /api/persist/disable` is sent when you disable caching
    * refresh the page after toggling the setting to make sure the change is saved
3. Ensure you can set the cache refresh interval for models
    * ensure the widget is completely hidden when caching is disabled
    * ensure you can select every option (1h, 2h, 3h, etc)
    * ensure `POST /api/persist/set-interval` with `{ hours: <int> }` is sent when you change the setting
    * refresh the page after changing the setting to make sure the change is saved
